### ISSUE TITLE
[typescript] Storage.getItem return type should not be any

### ISSUE THREAD

**Issue by Davilink** [quasarframework/quasar#5711]
**Is your feature request related to a problem? Please describe.**
the return type of Storage.getItem should not be 'any', because it hide the 'null' possibility.

**Describe the solution you'd like**
```ts
type ReturnType = Date | RegExp | number | boolean | string | object;

interface SessionStorage {
 // ...
 /**
     * Get a storage item value
     * @param key Entry key
     * @returns Storage item value
     */
    getItem (key : string): ReturnType | null;
// ...
}
```

**Additional context**
```ts
const jwt = storage.getItem(SessionStorageKey.Token);
```
instead of having this

![image](https://user-images.githubusercontent.com/1498932/69739387-a3de8780-1105-11ea-9b3a-a11aa0a4853a.png)

i would get 

![image](https://user-images.githubusercontent.com/1498932/69740441-64b13600-1107-11ea-9d73-48a0cffd1bef.png)

which clearly indicate that null is a possibility.

**Comment by panstromek** [comment]
@IlCallo you might be interested, not sure if you noticed already ;)

**Comment by IlCallo** [comment]
`this change is not as straightforward as it seems`

Can confirm this, current system doesn't allow to inject arbitrary TS types, I'm thinking a solution capable of covering this need

**Comment by rstoenescu** [comment]
Fix will be available in "quasar" v1.6.2.
